plugins/libplugin-pay.c

```diff
@@ -209,16 +210,80 @@ tal_createonion_response_from_json(const tal_t *ctx, const char *buffer,
 	return tal_free(resp);
 }
 
+static struct payment_result *tal_sendpay_result_from_json(const tal_t *ctx,
+						    const char *buffer,
+						    const jsmntok_t *toks)
+{
+	const jsmntok_t *idtok = json_get_member(buffer, toks, "id");
+	const jsmntok_t *hashtok = json_get_member(buffer, toks, "payment_hash");
+	const jsmntok_t *partidtok = json_get_member(buffer, toks, "partid");
+	const jsmntok_t *senttok = json_get_member(buffer, toks, "amount_sent_msat");
+	const jsmntok_t *statustok = json_get_member(buffer, toks, "status");
+	const jsmntok_t *preimagetok = json_get_member(buffer, toks, "payment_preimage");
+	const jsmntok_t *codetok = json_get_member(buffer, toks, "code");
+	const jsmntok_t *datatok = json_get_member(buffer, toks, "data");
+	struct payment_result *result;
+
+	/* Check if we have an error and need to descend into data to get
+	 * details. */
+	if (codetok != NULL && datatok != NULL) {
+		idtok = json_get_member(buffer, datatok, "id");
+		hashtok = json_get_member(buffer, datatok, "payment_hash");
+		partidtok = json_get_member(buffer, datatok, "partid");
+		senttok = json_get_member(buffer, datatok, "amount_sent_msat");
+		statustok = json_get_member(buffer, datatok, "status");
+	}
+
+	/* Initial sanity checks, all these fields must exist. */
+	if (idtok == NULL || idtok->type != JSMN_PRIMITIVE ||
+	    hashtok == NULL || hashtok->type != JSMN_STRING ||
+	    senttok == NULL || senttok->type != JSMN_STRING ||
+	    statustok == NULL || statustok->type != JSMN_STRING) {
+		fprintf(stderr, "XXX missing field\n");
```

Comment:
....

Reply:
Oops, removed in favor of logging in the caller.